Fix warnings in main

On M1
```
In file included from /Users/ondrej/repos/lpython/src/lpython/semantics/python_ast_to_asr.cpp:27:
/Users/ondrej/repos/lpython/src/lpython/semantics/python_comptime_eval.h:698:23: warning: unused variable 'arg_type' [-Wunused-variable]
        ASR::ttype_t *arg_type = ASRUtils::expr_type(arg);
                      ^
/Users/ondrej/repos/lpython/src/lpython/semantics/python_comptime_eval.h:715:23: warning: unused variable 'arg_type' [-Wunused-variable]
        ASR::ttype_t *arg_type = ASRUtils::expr_type(arg);
                      ^
/Users/ondrej/repos/lpython/src/lpython/semantics/python_comptime_eval.h:718:20: warning: comparison of integers of different signs: 'int' and 'std::basic_string<char>::size_type' (aka 'unsigned long') [-Wsign-compare]
        while (ind < res.size() && res[ind++] == ' ');
               ~~~ ^ ~~~~~~~~~~
/Users/ondrej/repos/lpython/src/lpython/semantics/python_comptime_eval.h:732:23: warning: unused variable 'arg_type' [-Wunused-variable]
        ASR::ttype_t *arg_type = ASRUtils::expr_type(arg);
                      ^
/Users/ondrej/repos/lpython/src/lpython/semantics/python_comptime_eval.h:735:18: warning: comparison of integers of different signs: 'int' and 'std::basic_string<char>::size_type' (aka 'unsigned long') [-Wsign-compare]
        while (l < res.size() && r >= 0 && (res[l] == ' ' || res[r] == ' ')) {
               ~ ^ ~~~~~~~~~~
/Users/ondrej/repos/lpython/src/lpython/semantics/python_ast_to_asr.cpp:4551:32: warning: comparison of integers of different signs: 'int' and 'std::basic_string<char>::size_type' (aka 'unsigned long') [-Wsign-compare]
                    while (ind < res.size() && res[ind] == ' ') {
                           ~~~ ^ ~~~~~~~~~~
/Users/ondrej/repos/lpython/src/lpython/semantics/python_ast_to_asr.cpp:4567:30: warning: comparison of integers of different signs: 'int' and 'std::basic_string<char>::size_type' (aka 'unsigned long') [-Wsign-compare]
                    while (l < res.size() && r >= 0 && (res[l] == ' ' || res[r] == ' ')) {
                           ~ ^ ~~~~~~~~~~
7 warnings generated.

```

--- a/src/lpython/semantics/python_ast_to_asr.cpp
+++ b/src/lpython/semantics/python_ast_to_asr.cpp
@@ -4585,7 +4585,7 @@ public:
                     }
                     AST::ConstantStr_t *n = AST::down_cast<AST::ConstantStr_t>(at->m_value);
                     std::string res = n->m_value;
-                    int ind = res.size()-1;
+                    int ind = (int)res.size() - 1;
                     while (ind >= 0 && res[ind] == ' '){
                         ind--;
                     }
@@ -4601,7 +4601,7 @@ public:
                     }
                     AST::ConstantStr_t *n = AST::down_cast<AST::ConstantStr_t>(at->m_value);
                     std::string res = n->m_value;
-                    int ind = 0;
+                    size_t ind = 0;
                     while (ind < res.size() && res[ind] == ' ') {
                         ind++;
                     }
@@ -4617,8 +4617,9 @@ public:
                     }
                     AST::ConstantStr_t *n = AST::down_cast<AST::ConstantStr_t>(at->m_value);
                     std::string res = n->m_value;
-                    int l = 0 ,r = res.size() - 1;
-                    while (l < res.size() && r >= 0 && (res[l] == ' ' || res[r] == ' ')) {
+                    size_t l = 0;
+                    int r = (int)res.size() - 1;
+                    while (l < res.size() && (int)r >= 0 && (res[l] == ' ' || res[r] == ' ')) {
                         l += res[l] == ' ';
                         r -= res[r] == ' ';
                     }

--- a/src/lpython/semantics/python_comptime_eval.h
+++ b/src/lpython/semantics/python_comptime_eval.h
@@ -695,9 +695,8 @@ struct PythonIntrinsicProcedures {
             throw SemanticError("str.rstrip() takes no arguments", loc);
         }
         ASR::expr_t *arg = args[0];
-        ASR::ttype_t *arg_type = ASRUtils::expr_type(arg);
         std::string res = ASR::down_cast<ASR::StringConstant_t>(arg)->m_s;
-         int ind = res.size()-1;
+        int ind = (int)res.size() - 1;
         while (ind >= 0 && res[ind--] == ' ');
         res = std::string(res.begin(), res.begin() + ind +1);
         ASR::ttype_t *type = ASRUtils::TYPE(ASR::make_Character_t(al, loc,
@@ -712,9 +711,8 @@ struct PythonIntrinsicProcedures {
             throw SemanticError("str.lstrip() takes no arguments", loc);
         }
         ASR::expr_t *arg = args[0];
-        ASR::ttype_t *arg_type = ASRUtils::expr_type(arg);
         std::string res = ASR::down_cast<ASR::StringConstant_t>(arg)->m_s;
-        int ind = 0;
+        size_t ind = 0;
         while (ind < res.size() && res[ind++] == ' ');
         res = std::string(res.begin() + ind, res.end());
         ASR::ttype_t *type = ASRUtils::TYPE(ASR::make_Character_t(al, loc,
@@ -729,9 +727,9 @@ struct PythonIntrinsicProcedures {
             throw SemanticError("str.strip() takes no arguments", loc);
         }
         ASR::expr_t *arg = args[0];
-        ASR::ttype_t *arg_type = ASRUtils::expr_type(arg);
         std::string res = ASR::down_cast<ASR::StringConstant_t>(arg)->m_s;
-        int l = 0 ,r = res.size() - 1;
+        size_t l = 0;
+        int r = (int)res.size() - 1;
         while (l < res.size() && r >= 0 && (res[l] == ' ' || res[r] == ' ')) {
             l += res[l] == ' ';
             r -= res[r] == ' ';

